### PR TITLE
resolve minor issues in k8s secret integration with CPA

### PR DESCRIPTION
**Pull request by shenmo3**
refactor code and import and messages that are left when implementing cloud credentials to use k8s secret.

Signed-off-by: Alexander Liu <alliu@vmware.com>

### CODE REVIEW

File: pkg/cloud-provider/cloudapi/azure/azure_account_mgmt.go [hunk 1]
```diff
@@ -80,10 +79,6 @@ func compareAccountCredentials(accountName string, existing interface{}, new int
 
 // extractSecret extracts credentials from a Kubernetes secret.
 func extractSecret(c client.Client, s *v1alpha1.SecretReference) (*v1alpha1.AzureAccountCredential, error) {
-	if s == nil {
-		return nil, fmt.Errorf("secret reference not found")
```
Comment (bangqipropel) [inline pkg/cloud-provider/cloudapi/azure/azure_account_mgmt.go:84]: One thing here, we have a lot of error messages same in Azure and AWS here, like this `secret reference not found` , shall we add some suffix here to distinguish? maybe "secret reference not found in Azure" and secret reference not found in AWS" may be better?

File: config/samples/cloudaccount_aws.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Add AWS Account and Onboard VPC
 # To get base64 encoded json string for secret credential, run:
-# echo '{"accessKeyId": "YOUR_AWS_ACCESS_KEY_ID", "accessKeySecret": "YOUR_AWS_ACCESS_KEY_SECRET", "roleArn": "YOUR_AWS_IAM_ROLE_ARN", "externalId": "IAM_ROLE_EXTERNAL_ID"}' | openssl base64
+# echo '{"accessKeyId": "YOUR_AWS_ACCESS_KEY_ID", "accessKeySecret": "YOUR_AWS_ACCESS_KEY_SECRET", "roleArn": "YOUR_AWS_IAM_ROLE_ARN",
+# "externalId": "IAM_ROLE_EXTERNAL_ID"}' | openssl base64
```
Comment (reachjainrahul) [inline config/samples/cloudaccount_aws.yaml:4]: This might cause JSON parser error as discussed.

Comment (shenmo3) [reply config/samples/cloudaccount_aws.yaml]: Done

File: pkg/cloud-provider/cloudapi/azure/azure_account_mgmt.go (hunk 1 above)
Comment (shenmo3) [reply pkg/cloud-provider/cloudapi/azure/azure_account_mgmt.go:84]: As discussed offline, the errors will also print out stacktraces, so its not really hard to identify whether aws or azure is having problem

**Review by reachjainrahul — APPROVED** [review]

**Review by reachjainrahul — APPROVED** [review]

**Pull request merged**